### PR TITLE
added loading state in first page load

### PR DESCRIPTION
**Pull request by uxitten**
sometime server is delay to response or internet connection is low

### CODE REVIEW

**Comment by krilnon** [comment]
Why add `debounce`? I don't think that rendering while typing in the filter box poses a performance issue.

**Comment by uxitten** [comment]
@krilnon yes i agree with you, but if there's a big record, it's a problem

**Comment by tkremenek** [comment]
@krilnon you OK with merging this?

File: index.html [hunk 1]
```diff
@@ -77,7 +77,7 @@ <h1 id="title"><a href="#">Swift Evolution</a></h1>
                 <article class="proposals">
                     <div id="proposals-count"><span id="proposals-count-number"></span></div>
                     <ul class="proposals-list">
-                        <!-- {proposals} -->
+                        <!-- {proposals} -->
```
Comment (krilnon) [inline index.html]: Why add whitespace here?

File: index.js [hunk 2]
```diff
@@ -128,9 +128,10 @@ function init () {
   })
 
   req.addEventListener('error', function (e) {
-    document.querySelector('#proposals-count').innerText = 'Proposal data failed to load.'
+    document.querySelector('#proposals-count-number').innerText = 'Proposal data failed to load.'
   })
 
+  document.querySelector('#proposals-count-number').innerHTML = 'Loading ...';
```
Comment (krilnon) [inline index.js]: Remove the semicolon here and other places where they appear to follow the [style guide](https://standardjs.com).

File: index.html (hunk 1 above)
Comment (uxitten) [reply index.html]: thanks, fixed

File: index.js (hunk 2 above)
Comment (uxitten) [reply index.js]: fixed

**Comment by uxitten** [comment]
i think we need `debounce` because the volume of data will increase in the future

**Comment by uxitten** [comment]
is not this resolved?

**Comment by tkremenek** [comment]
@krilnon?

**Comment by krilnon** [comment]
There are still some style guide rules being broken here. You fixed the two places with semicolons that I pointed out explicitly, but you should just remove all the ones you added. There's [a tool to check these rules automatically](https://standardjs.com/#usage) that you might find useful.

> i think we need `debounce` because the volume of data will increase in the future

Is there a place where you've seen a performance issue here, or, if you haven't seen one yet, have you done any measurements that indicate *the conditions in which* we'd need debounce? I'm guessing it'll be a couple more years until we even approach 1000 proposals; and I don't think there'd be performance problems even then.

**Comment by uxitten** [comment]
@krilnon all right, i am removing debounce

**Comment by uxitten** [comment]
@krilnon removed now

**Comment by krilnon** [comment]
Alright, looks good now. Thanks for the improvement and for working through this with me!

**Comment by uxitten** [comment]
@krilnon you're welcome

**Pull request merged**